[DOCS] Clarify Expectation Terminology, Fix Links, Improve Distributional Explanations (#1872)
great-expectations/great_expectations: docs/reference/core_concepts/expectations/distributional_expectations.rst

Partition Objects
--------------------------------------------------------------------------------

> The core constructs of a great expectations distributional expectation are the partitions, which are intervals/classes in a distributions and are made of histogram bins and their associated weights.

Reviewer: ```suggestion
Distributional expectations rely on expected distributions or "partition objects", which are built from intervals for continuous data or categorical classes and their associated weights.
```

* However, following the behavior of numpy.histogram, a data element x is in the largest bin k if x == upper_bound_k.
* A partition object can also include ``tail_weights`` which extend from -Infinity to the lowest bound, and from the highest bound to +Infinity.

> * Partition weights define the probability of the associated bin interval. Note that this applies a "piecewise uniform" distribution to the data for the purpose of statistical tests. The weights must define a valid probability distribution, i.e. they must be non-negative numbers that sum to 1.

Reviewer: ```suggestion
* Partition weights define the probability of the associated bin or interval. Note that this applies a "piecewise uniform" distribution to the data for the purpose of statistical tests. The weights must define a valid probability distribution, i.e. they must be non-negative numbers that sum to 1.
```

Available Distributional Expectations
--------------------------------------

> `Kullback-Leibler (KL) divergence <https://www.youtube.com/watch?v=ErfnhcEV1O8/>`_ is available as an expectation for both categorical and continuous data (continuous data will be discretized according to the provided partition prior to computing divergence). Unlike KS and Chi-Squared tests which can use a p-value, you must provide a threshold for the relative entropy to use KL divergence. Further, KL divergence is not symmetric.

Reviewer: ```suggestion
Kullback-Leibler (KL) divergence (also known as `relative entropy https://www.youtube.com/watch?v=ErfnhcEV1O8`_) is available as an expectation for both categorical and continuous data (continuous data will be discretized according to the provided partition prior to computing divergence). Unlike KS and Chi-Squared tests which can use a p-value, you must provide a threshold for the relative entropy to use KL divergence. Further, KL divergence is not symmetric.
```
